### PR TITLE
Tidy nodes so we can call their functionality

### PR DESCRIPTION
**Pull request by nigelmegitt**
This way we can ask the nodes to do what they do without creating carriage mechanisms, if we want to manage data flow manually.

### CODE REVIEW

File: ebu_tt_live/node/denester.py [hunk 1]
```diff
@@ -1,28 +1,71 @@
...
+                log.warning(
+                    'Ignoring duplicate document: {}__{}'.format(
+                        document.sequence_identifier,
+                        document.sequence_number
+                    )
+                )
+        else:
+            self.producer_carriage.emit_data(data=document, **kwargs)
+
+    def denest(document):
```
Comment (danielthepope) [inline ebu_tt_live/node/denester.py]: Shouldn't all these functions have `self` as their first argument? Alternatively you should be able to move the function declarations out of the class definition.

Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: It seems like a philosophical question. The functions are gathered into a class for convenience and addressing, but none of them needs access to `self` so why make them have it? I'd be more strongly in favour of a change if there were some data that needed to be stored in the object and accessed in the functions, but I couldn't see any!

Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: By the way, they don't even need `cls`, either.

Comment (danielthepope) [reply ebu_tt_live/node/denester.py]: Looking at another e.g. `delay.py`, it has functions declared outside the class. They'll still be in the same python file so it's still easy to reference.

It just seems odd having all these warnings pop up on my screen, makes it seem like this sort of pattern isn't standard practice even though it seems to work.

Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: Discussion result: add `@staticmethod` decorator to all the static methods.

Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: Fixed in f465381

File: ebu_tt_live/documents/ebuttd.py [hunk 2]
```diff
@@ -24,7 +24,16 @@ def __init__(self, lang):
...
+    @property
+    def binding(self):
+        return self._ebuttd_content
+
+    @property
+    def implicit_ns(self):
+        return self._implicit_ns
+
+    @implicit_ns.setter
+    def implicit_ns(self, value):
```
Comment (danielthepope) [inline ebu_tt_live/documents/ebuttd.py]: Ah yes, this is a better way of doing Python-like getters and setters 👍

File: ebu_tt_live/node/denester.py [hunk 3]
```diff
@@ -197,73 +302,102 @@ def recurse(div, dataset, merged_attr={ "styles": [], "begin": None, "end": None
 
                 new_div = div_type(
                     id=div.id,
-                    style=None if len(merged_attr["styles"]) == 0  else merged_attr["styles"],
-                    begin=DenesterNode.process_timing_from_timedelta(merged_attr["begin"]) if merged_attr["begin"] is not None else merged_attr["begin"],
-                    end=DenesterNode.process_timing_from_timedelta(merged_attr["end"]) if merged_attr["end"] is not None else merged_attr["end"],
+                    style=None
+                    if len(merged_attr["styles"]) == 0
+                    else merged_attr["styles"],
```
[the remark this answers is not in the document]
Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: I know, it's a pain, and when I tried indenting the `if` and `else` flake8 got unhappy.

[the remark this answers is not in the document]
Comment (nigelmegitt) [reply ebu_tt_live/node/denester.py]: Shall we just do nothing about this for the time being?

File: ebu_tt_live/bindings/__init__.py [hunk 4]
```diff
@@ -1480,7 +1482,7 @@ def _validateBinding_vx(self):
         raw.layout: layout,
         raw.body_type: body_type,
     },
-}
+    }
```
[the remark this answers is not in the document]
Comment (nigelmegitt) [reply ebu_tt_live/bindings/__init__.py]: Fixed in c7e261c .

File: ebu_tt_live/node/denester.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielthepope) [reply ebu_tt_live/node/denester.py]: Yeah, alright :)

**Pull request merged**